compiler/noirc_frontend/src/node_interner.rs

```diff
@@ -1282,6 +1280,17 @@ impl NodeInterner {
         self.find_location_index(location)
             .and_then(|index| self.resolve_location(index))
             .or_else(|| self.try_resolve_trait_impl_location(location))
+            .or_else(|| self.try_resolve_trait_method_declaration(location))
+    }
+
+    pub fn get_declaration_location_from(&self, location: Location) -> Option<Location> {
+        self.try_resolve_trait_method_declaration(location).or_else(|| {
+            self.find_location_index(location)
+                .and_then(|index| self.resolve_location(index))
+                .and_then(|found_impl_location| {
+                    self.try_resolve_trait_method_declaration(found_impl_location)
+                })
+        })
```

Comment:
This looks good but now that the number of location resolution functions are increasing, can we move them to a dedicated file? `node_interner.rs` is getting quite long.

Reply:
d941309: chore: separate location resolution to own file